### ISSUE TITLE
MD050 Regression

### ISSUE THREAD

**Issue by mvilrokx** [DavidAnson/markdownlint#1141]
# Problem
When I use `v0.27.0` (via `markdownlint-cli2` `v0.6.0`) the following is raised as an error:

```text
Summary: 1 error(s)
...md:173:80 MD050/strong-style Strong style should be consistent [Expected: underscore; Actual: asterisk]
```

When I use `v0.28.1` (via `markdownlint-cli2` `v0.7.0`) the following are raised as errors:

```text
Summary: 1 error(s)
...md:173:80 MD050/strong-style Strong style should be consistent [Expected: underscore; Actual: asterisk]
```

When I use `v0.29.0` (via `markdownlint-cli2` `v0.8.0`) _or later_, **no errors** are raised.

## `MD050` Regression
The `MD050` omission is definitely a regression because it should get reported as a linting error:

```markdown
- The deployment __kube-state-metrics__ must have a replica count of 1 for the **kube-state-metrics**
```
As you can see, the last strong word (`**kube-state-metrics**`) should get flagged.

This was caught because our build-platform uses an older version (not under our control) but our GHA is using the latest version.

**Comment by DavidAnson** [comment]
I think you are reporting the following bug where the very first or last word with improper emphasis will not be flagged: https://github.com/DavidAnson/markdownlint/issues/1132

If so, I have already fixed this for the next release. Sorry about that!

**Comment by mvilrokx** [comment]
Possible, I will test when it gets released 👍.

**Comment by DavidAnson** [comment]
It looks like the same thing. You can see that even adding a period at the end is enough to get the second emphasis flagged: https://dlaa.me/markdownlint/#%25m%23%20Issue%201141%0A%0A-%20The%20deployment%20__kube-state-metrics__%20must%20have%20a%20replica%20count%20of%201%20for%20the%20**kube-state-metrics**.%0A

**Comment by DavidAnson** [comment]
Confirmed duplicate of https://github.com/DavidAnson/markdownlint/issues/1132 and fixed in `next` branch.
